### PR TITLE
Hooks 🎣

### PR DESCRIPTION
**Pull request by alexreardon**
#871

### CODE REVIEW

File: .size-snapshot.json [hunk 1]
```diff
@@ -1,25 +1,25 @@
...
+    "minified": 147989,
+    "gzipped": 41340
   },
   "dist/react-beautiful-dnd.min.js": {
-    "bundled": 303784,
-    "minified": 113686,
-    "gzipped": 32910
+    "bundled": 327193,
+    "minified": 117669,
+    "gzipped": 33593
```
Comment (alexreardon) [inline .size-snapshot.json]: This was suprising

File: .size-snapshot.json [hunk 2]
```diff
@@ -1,25 +1,25 @@
...
-    "minified": 125718,
-    "gzipped": 31356,
+    "bundled": 239473,
+    "minified": 123943,
+    "gzipped": 31513,
     "treeshaked": {
       "rollup": {
-        "code": 85891,
-        "import_statements": 832
+        "code": 30284,
```
Comment (alexreardon) [inline .size-snapshot.json]: This is quite a nice drop. @TrySound I am not sure what this property is referring too. Can you please explain this property?

File: .size-snapshot.json (hunk 1 above)
Comment (alexreardon) [reply .size-snapshot.json]: I was expecting moving away from classes would be a drop, but it has gone the other way

File: src/view/droppable/droppable.jsx [hunk 3]
```diff
@@ -1,191 +1,145 @@
...
+          innerRef={setPlaceholderRef}
+          animate={animate}
+          styleContext={styleContext}
+          onTransitionEnd={onPlaceholderTransitionEnd}
+        />
+      )}
+    </AnimateInOut>
+  );
+
+  // const placeholder: Node | null = instruction ? (
```
Comment (alexreardon) [inline src/view/droppable/droppable.jsx]: TODO: remove

File: src/view/placeholder/placeholder.jsx [hunk 4]
```diff
@@ -48,129 +58,145 @@ const empty: Size = {
...
+    marginLeft: size.margin.left,
+
+    // ## Avoiding collapsing
+    // Avoiding the collapsing or growing of this element when pushed by flex child siblings.
+    // We have already taken a snapshot the current dimensions we do not want this element
+    // to recalculate its dimensions
+    // It is okay for these properties to be applied on elements that are not flex children
+    flexShrink: '0',
+    flexGrow: '0',
+    // Just a little performance optimisation: avoiding the browser needing
```
Comment (alexreardon) [inline src/view/placeholder/placeholder.jsx]: TODO: move to style marshal?

File: .size-snapshot.json (hunk 2 above)
Comment (TrySound) [reply .size-snapshot.json]: This is the code in user bundle when you import nothing. Useful for ui libs like material ui. Does not mean a lot for this lib though.

File: src/view/drag-drop-context/use-startup-validation.js [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+import checkDoctype from './check-doctype';
+
+export default function useStartupValidation() {
+  // Only need to run these when mounting
+  useEffect(() => {
+    if (process.env.NODE_ENV === 'production') {
+      return;
+    }
+
+    checkReactVersion(peerDependencies.react, React.version);
```
Comment (TrySound) [inline src/view/drag-drop-context/use-startup-validation.js]: I would wrap this check with condition. It looks like early returns does not eliminated well with webpack uglify pair.

Comment (alexreardon) [reply src/view/drag-drop-context/use-startup-validation.js]: Thanks

Comment (alexreardon) [reply src/view/drag-drop-context/use-startup-validation.js]: TODO: update

**Pull request merged**